src/particle.cpp

```diff
@@ -203,6 +203,11 @@ void Particle::event_advance()
   }
   this->time() += distance / this->speed();
 
+  // Kill particle if its time exceeds the cutoff
+  if (time() > settings::time_cutoff[static_cast<int>(type())]) {
+    wgt() = 0.0;
+  }
```

Comment:
In both MCNP and Serpent, I believe it calculates the distance to the time cutoff and won't let the particle move past that time -- here we're not really stopping the particle at the time cutoff and in principle it could go much past the time specified, which makes me worried and seems like it could create surprising behavior for the user.

Reply:
The documentation does clarify the expected behavior here. Despite this, I agree with you Paul. It costs nearly the same arithmetic to check along the pathlength rather than at collision sites, so why not take this more precise approach?